### PR TITLE
DM-40873: make nbexec return more useful data on notebook error

### PR DESCRIPTION
**Pull request by athornton**
Always return an object from the notebook execution, and if there's an error, indicate which cell and give a traceback.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -110,3 +110,6 @@ dmypy.json
 
 # OSX files
 .DS_Store
+
+# Package lock?  May want to remove this
+package-lock.json
```
Comment (jonathansick) [inline .gitignore]: Normally you want to keep the package-lock.json in version control for reproducible builds.

**Review by jonathansick — APPROVED** [review]
This looks great, thanks!

**Comment by jonathansick** [comment]
The CI is failing because it's set up to run tox, but this project doesn't use tox to run the tests?

**Comment by athornton** [comment]
Yeah, this is (an old) jupyter cookiecutter recipe.  Doesn't natively use tox, pre-commit, typing, any of our usual stuff.  And some of that's quite valuable.

Likewise with package.json.  It's as of yet unclear to me whether I can get this happy as a prebuilt extension or whether I'm going to need to run jupyter lab build.  If the second then I definitely want all the input packages from the existing jupyterlab incarnation rather than whatever I built it with standalone.

**Comment by jonathansick** [comment]
Since tox.ini is only used for development, it doesn't need to be included in the packaging via `MANIFEST.in`.

**Pull request merged**